sgt-back-end/index.js

```diff
@@ -0,0 +1,188 @@
+const express = require('express');
+const app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+    from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const { name, course, score } = req.body;
+  if (!req.body.name) {
+    res.status(400).json({
+      error: 'name is a required field'
+    });
+    return;
+  } else if (!req.body.course) {
+    res.status(400).json({
+      error: 'course is a required field'
+    });
+    return;
+  } else if (!Number.isInteger(score) || score <= 0) {
+    res.status(400).json({
+      error: 'score must be a positive integer'
+    });
+    return;
+  }
```

Comment:
Your allowed values for scores are off:

---

![image](https://user-images.githubusercontent.com/16143771/186958179-29bb4a53-43fe-48fd-b2eb-e2ecc6016ad0.png)